### PR TITLE
Stability fix and small refactoring

### PR DESCRIPTION
**Pull request by arkpar**
Fixes possible  corruption on critical write error.
Also include a small refactoring.

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -38,18 +38,24 @@ use fs2::FileExt;
...
 	index::PlanOutcome,
 	options::Options,
 	btree::{BTreeIterator, BTreeTable, commit_overlay::{BTreeChangeSet}},
 };
 
+// Max size of commit queue. (Keys + Values). If the queue is
+// full `commit` will block.
 // These are in memory, so we use usize
 const MAX_COMMIT_QUEUE_BYTES: usize = 16 * 1024 * 1024;
 // These are disk-backed, so we use u64
```
Comment (cheme) [inline src/db.rs:52]:
```suggestion

```

**Review by cheme — APPROVED** [review]
Looks good to me.

File: src/index.rs [hunk 2]
```diff
@@ -485,7 +498,8 @@ impl IndexTable {
 
 	pub fn flush(&self) -> Result<()> {
 		if let Some(map) = &*self.map.read() {
-			map.flush()?;
+			// Flush everything except stats.
```
[the remark this answers is not in the document]
Comment (arkpar) [reply src/index.rs:501]: Stats are update on each write. If there are a lot of commits that only touch a few index pages flushing stats each time can be expensive.

File: src/column.rs [hunk 3]
```diff
@@ -114,6 +114,25 @@ enum IterStateOrCorrupted {
 	Corrupted(crate::index::Entry, Option<Error>),
 }
 
+#[inline]
+pub fn hash_key(key: &[u8], salt: &Salt, uniform: bool, db_version: u32) -> Key {
+	let mut k = Key::default();
+	if uniform {
+		if db_version <= 5 {
```
[the remark this answers is not in the document]
Comment (arkpar) [reply src/column.rs:121]: Yeah, searching for `db_version` is what I'd do

**Pull request merged**